tidb-in-kubernetes/k8s-deployment.md

```diff
@@ -0,0 +1,99 @@
+---
+title: 部署 k8s
+category: how-to
+---
+
+# 软件版本要求
+
+| 软件名称 | 版本 |
+| --- | --- |
+| Docker | docker-ce 18.09.6 |
+| K8s |  v1.12.2 |
+| Linux 系统 | 7.3 或以上 |
+
+# 内核参数设置
+| 配置项 | 设置值 |
+| --- | --- |
+| net.core.somaxconn | 32768 |
+| vm.swappiness | 0 |
+| net.ipv4.tcp_syncookies | 1 |
+| net.ipv4.ip_forward | 1 |
+| fs.file-max | 1000000 |
+| fs.inotify.max_user_watches | 1048576 |
+| fs.inotify.max_user_instances | 1024 |
+| net.ipv4.neigh.default.gc_thresh1 | 80000 |
+| net.ipv4.neigh.default.gc_thresh2 | 90000 |
+| net.ipv4.neigh.default.gc_thresh3 | 100000 |
+| net.bridge.bridge-nf-call-iptables | 1 |
+| net.bridge.bridge-nf-call-arptables | 1 |
+| net.bridge.bridge-nf-call-ip6tables | 1 |
+
+在设置 `net.bridge.bridge-nf-call-*` 这几项参数时如果报选项不存在错误，通过如下命令检查是否已经加载改模块：
+
+```shell
+$ ls mod|grep br_netfilter
+```
+如果没有加载则执行如下命令加载：
+
+```shell
+$ modprobe br_netfilter
+```
+
+同时还需要关闭每个部署 k8s 节点的 swap，执行命令如下：
+
+```shell
+$ swapofff -a
+```
+
+执行如下命令检查 swap 已经关闭：
+
+```shell
+## 执行命令后输出显示 swap 一列全是 0 则表明 swap 已经关闭
+$ free -m
+```
+
+# 硬件和部署要求
+
+和部署 tidb binary 集群的服务器要求一致，选用 Intel x86-64 架构的 64 位通用硬件服务器，网卡使用万兆网卡。对于服务器 disk、memory、cpu 的选择要看对集群的容量规划以及部署拓扑来定。线上 k8s 集群部署为了保证搞可用一般需要部署三个 master 节点，三个 etcd 节点，若干个 work 节点，同时为了充分的利用机器资源，master 节点一般也充当 work 节点(也就是 master 节点上也可以调度负载)，通过 kubelet 设置 。[预留资源](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/) 来保证机器上的系统进程以及 k8s 的核心进程在工作负载很高的情况下仍然有足够的资源来运行，从而保证整个系统的稳定。下面按 3 k8s master + 3 etcd + 若干 work 节点部署方案分析。
```

Comment:
Add a link to TiDB deployment requirement document.
And add a reference link to k8s document about how to deploy HA cluster.